### ISSUE TITLE
Unity Cardboard XR - Black Screen - iOS 13.5 & Android 10 - Unity 2019.3.14

### ISSUE THREAD

**Issue by andrewio** [googlevr/cardboard#46]
Buttons: Gear & Close are not working.
No Permission Request Window. App Restart doesn't help.
Unity 2019.13.4

All Points here are executed: https://developers.google.com/cardboard/develop/unity/quickstart

Here is iOS log.

```
2020-05-26 15:54:54.107680+0300 testk[1983:423140] Built from '2019.3/release' branch, Version '2019.3.14f1 (2b330bf6d2d8)', Build type 'Release', Scripting Backend 'il2cpp'
-> applicationDidFinishLaunching()
-> applicationDidBecomeActive()
[Subsystems] Discovering subsystems at path /private/var/containers/Bundle/Application/123/testk.app/Data/UnitySubsystems
[Subsystems] No descriptors matched for  examples in UnitySubsystems/CardboardXrProvider/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/CardboardXrProvider/UnitySubsystemsManifest.json
[Subsystems] 1 'displays' descriptors matched in UnitySubsystems/CardboardXrProvider/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  meshings in UnitySubsystems/CardboardXrProvider/UnitySubsystemsManifest.json.
GfxDevice: creating device client; threaded=1
Renderer: Apple A12 GPU
Vendor:   Apple Inc.
Version:  OpenGL ES 2.0 Metal - 66.6
GLES:     2
 GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_OES_depth24 GL_OES_element_index_uint GL_OES_fbo_render_mipmap GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_float GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_vertex_array_object GL_EXT_blend_minmax GL_EXT_color_buffer_half_float GL_EXT_debug_label GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_draw_instanced GL_EXT_instanced_arrays GL_EXT_map_buffer_range GL_EXT_occlusion_query_boolean GL_EXT_pvrtc_sRGB GL_EXT_read_format_bgra GL_EXT_separate_shader_objects GL_EXT_shader_framebuffer_fetch GL_EXT_shader_texture_lod GL_EXT_shadow_samplers GL_EXT_sRGB GL_EXT_texture_filter_anisotropic GL_EXT_texture_rg GL_EXT_texture_storage GL_APPLE_clip_distance GL_APPLE_color_buffer_packed_float GL_APPLE_copy_texture_levels GL_APPLE_framebuffer_multisample GL_APPLE_rgb_422 GL_APPLE_sync GL_APPLE_texture_format_BGRA8888 GL_APPLE_texture_max_level GL_APPLE_texture_packed_float GL_IMG_
read_format GL_IMG_texture_compression_pvrtc
OPENGL LOG: Creating OpenGL ES 2.0 graphics device ; Context level  <OpenGL ES 2.0> ; Context handle -2119318528
OPENGL LOG: OpenGLES2 is deprecated on this platform
Initialize engine version: 2019.3.14f1 (2b330bf6d2d8)
XRGeneral Settings awakening...
UnityEngine.XR.Management.XRGeneralSettings:Awake()
 
(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

[Subsystems] Loading plugin CardboardXrProvider for subsystem Display...
[Subsystems] Cardboard successfully registered Provider for Display
[XR] [CardboardXrMain]: Display subsystem successfully loaded.
[Subsystems] Cardboard successfully registered Provider for Input
[Subsystems] Cardboard successfully registered Provider for Display
[XR] [CardboardXrMain]: Input subsystem successfully loaded.
[Subsystems] Loading plugin CardboardXrProvider for subsystem Input...
[XR] [CardboardXrInputProvider]: Lifecycle initialized
[XR] [CardboardXrInputProvider]: Lifecycle started
[XR] [CardboardXrDisplayProvider]: Initializes Cardboard API.
RequestCreateTexture
RequestCreateTexture
SetupRenderTextureFromXRRequest (id: 1 col: 0x14 d: 0x3)
SetupRenderTextureFromXRRequest (id: 2 col: 0x15 d: 0x4)
RenderTexture->Create
RenderTexture->Create
IVRDeviceCallback_CreateEyeTextureResources 0x14
IVRDeviceCallback_CreateEyeTextureResources 0x15
2020-05-26 15:54:55.433516+0300 testk[1983:423140] Unbalanced calls to begin/end appearance transitions for <UIViewController: 0x10cf0d190>.
```

**Comment by grzegorz-gornisiewicz** [comment]
Sadly OpenGLES2 & OpenGLES3 are deprecated for iOS.
![Zrzut ekranu 2020-05-28 o 01 13 04](https://user-images.githubusercontent.com/308661/83081307-a9029c00-a080-11ea-8183-7ea96edde28d.png)

**Comment by chusmaverde** [comment]
Hello, I am running into the same issue in a Samsung Galaxy S9 (Android). I just followed the [tutorial](https://developers.google.com/cardboard/develop/unity/quickstart), step by step. 

I build the HelloCardboard scene and all I get is a Black screen with the close & cog buttons (unresponsive) and a line in the middle of the screen.

Where can the problem be? I have checked the logs in "C:\Users\username\AppData\Local\Unity\Editor.log" and have not seen anything relevant. My Unity version is 2019.3.13f1

**Comment by jballoffet** [comment]
I executed the steps described above and I wasn’t able to reproduce this issue. Could you please provide a link to the Unity project that you’re building in order for me to test it?

**Comment by chusmaverde** [comment]
Good afternoon,
Thank you for looking into this. I have uploaded the project to [this location](https://www.dropbox.com/s/dxl54e4j5vk9j5k/VR%20Google%20CardBoard.7z?dl=0). It also includes the build I did for the android system in the Build folder.

As mentioned before, I followed the instructions from the official website. The only part I was not sure was about the mainTemplate.gradle modification, everything else looked straightforward.

My android system settings are:
Android version: 10
Baseband version: G960FXXS9DTD7
Kernel version: 4.9.118-18339707

If you need any other information (or there are problems with the compressed 7zip file) do not hesitate to ask me.

**Comment by developers412** [comment]
@chusmaverde @jballoffet am facing the same issue and am deploying on a samsung s9+ 
from unity 2019.4.2.f1

it's just a new fresh project with the sample scene mentioned in the guide [here](https://developers.google.com/cardboard/develop/unity/quickstart)
and here is a [link ](https://drive.google.com/drive/folders/13GPdSttPVPgt2Yz2zrld67XNsDZaqxLu?usp=sharing)for the project at the moment

**Comment by jballoffet** [comment]
Sorry for the long delay here. Thank you for the repro project, this issue has now been prioritized and we are working on it now. We'll update this issue once the fix is found.

**Comment by agalbachicar** [comment]
@andrewio I'm sorry but I could not reproduce your error.

@chusmaverde I tried your sample as it was but with Unity 2019.4.15f1 and just worked for me. No black screen was noticeable.

@developers412 I tried your sample but with Unity 2019.4.15f1 and yes, a black screen was noticed. I checked the project configuration and found that OpenGL ES3 was selected instead of OpenGL ES2. When made that change, everything worked. I checked logcat output with OpenGL ES3, and it said: `... cardboard_xr_unity.cc:572] GL error: 1282` (a failure). Note that you are using version 1.1 of the plugin which did not support OpenGL ES3.

I suggest to all the following verification steps:

- Download the latest version of Unity.
- Download the latest Cardboard XR Plugin
- Verify if the problem persists for you and reopen the ticket if it's related or open a new one if you encounter another problem.
